### PR TITLE
Feat/quiz

### PR DESCRIPTION
**Pull request by abhijitdotsharma**
Quiz question functionality added

- user can choose from different **Categories** in `CategoryPage` and it'll render a list of **questions** in `QuizPage`
- Only working logic added
- Styling and Routing to be done later

Files to check 
- `src/pages/category-page` with `src/components/Category.jsx`
- `src/pages/quiz-page`

How to test the PR
- In the landing page, click `Lets Go` -> click on `play now`

[Live Preview](https://deploy-preview-4--popcorn-quiz.netlify.app/)

### CODE REVIEW

**Comment by abhijitdotsharma** [comment]
p.s - I made an error by commiting some files related to this PR in an early PR, I'll try to learn more about how to change that if possible. And not repeat that, sorry for the inconvenience to the reviewer's

File: src/components/category/Category.jsx [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+        const response = await axios.get(`/api/categories/quizzes/${categoryId}`)
+        console.log(response.data.quizzes[0].mcqs)
+
+        //setMcqs to quizContext
+        setMcqs(prev => ({...prev, questions: response?.data?.quizzes[0]?.mcqs}))
+        navigate("/quiz-page")
     }
 
-    console.log("category comp - ", category)
+    // console.log("category comp - ", category)
```
Comment (t007rushi) [inline src/components/category/Category.jsx:23]: if not required can remove comments

File: src/context/quiz-context.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const QuizProvider = ({children}) => {
+
+
+    const [mcqs, setMcqs] = useState({
+        step: 0,
+        // step: 0,
+        // questions: [
+        // { id: 1, title: 'What is your name?' },
+        // { id: 2, title: 'What do you do?' },
+        // { id: 3, title: 'Where you from?' }
```
Comment (t007rushi) [inline src/context/quiz-context.js:17]: same as above

File: src/pages/home-page/HomePage.jsx [hunk 3]
```diff
@@ -1,57 +1,41 @@
...
+                    Lorem ipsum dolor, sit amet consectetur adipisicing elit.
+                </p>
+
+                <p className="txt-md">
+                    Test your football knowledge
+                </p>
+                <Link 
+                to="/category-page"
+                className="btn-landing-text"
+                onClick={() => console.log("test")}
```
Comment (t007rushi) [inline src/pages/home-page/HomePage.jsx:23]: console here

File: src/pages/quiz/QuizPage.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            >A. {mcqs.questions[mcqs.step].options[0]}</p>
+            <p
+                onClick={() => checkAnswer(mcqs.questions[mcqs.step].answer, mcqs.questions[mcqs.step].options[1])}
+            >B. {mcqs.questions[mcqs.step].options[1]}</p>
+            <p
+                onClick={() => checkAnswer(mcqs.questions[mcqs.step].answer, mcqs.questions[mcqs.step].options[2])}
+            >C. {mcqs.questions[mcqs.step].options[2]}</p>
+            <p
+                onClick={() => checkAnswer(mcqs.questions[mcqs.step].answer, mcqs.questions[mcqs.step].options[3])}
+            >D. {mcqs.questions[mcqs.step].options[3]}</p>
```
Comment (t007rushi) [inline src/pages/quiz/QuizPage.jsx:72]: this is repeating and only index changing can extract out logic outside
passing props and checks also

**Review by t007rushi — APPROVED** [review]
Looks good overall structure
work on some points else
Good to go

**Pull request merged**